### PR TITLE
Tabular: Fixed incorrect metric score calculation in evaluate

### LINKED ISSUE #565: Is the calculation of the RMSE of test data correct?
Is the RMSE calculation in evaluate_prediction in autogluon version 0.0.11 correct?

From a prediction with the test data:
```
y_pred = predictor.predict(X_raw)
perf = predictor.evaluate_predictions(
        y_true=y_test, y_pred=y_pred, auxiliary_metrics=True
    )
```
I got:

```
Evaluation: root_mean_squared_error on test data: 1.9002197479504435
Evaluations on test data:
{
    "root_mean_squared_error": 1.9002197479504435,
    "mean_absolute_error": 0.2822201467381229,
    "explained_variance_score": 0.9041746987036652,
    "r2_score": 0.9041688612988515,
    "pearson_correlation": 0.9509362562666829,
    "mean_squared_error": 0.17589136840340483,
    "median_absolute_error": 0.18991703222021486
}
```

When the RMSE is calculated using
```
from sklearn.metrics import mean_squared_error, r2_score
import numpy as np

def rmse(y_pred, y_true):
    return np.sqrt(mean_squared_error(y_pred, y_true)) 

print(f'RMSE: {rmse(y_test, y_pred):.3f}')
print(f'MSE: {mean_squared_error(y_test, y_pred):.3f}')
print(f'r2 score: {r2_score(y_test, y_pred):.3f}')
```
the following values are returned:
```
RMSE: 0.419
MSE: 0.176
r2 score: 0.904
```

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*
Resolves #565 
#544 

*Description of changes:*

- Fixed defect where ndarray y_pred and series y_true were sent as input to `predictor.evaluate_predictions` and y_true's index was not reset causing index misalignment and incorrect scoring of eval_metric.
- Improved code clarity / quality + added TODO's for future improvements
- Improved return value of confusion matrix to be a DataFrame instead of a json string.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-590-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-590/1/index.html
